include/CppUTest/TestMemoryAllocator.h

```diff
@@ -96,5 +96,37 @@ class NullUnknownAllocator: public TestMemoryAllocator
     static TestMemoryAllocator* defaultAllocator();
 };
 
+
+struct LocationToFailAlloc
+{
+    int allocNumberToFail;
+    int actualAllocNumber;
+    const char* file;
+    int line;
+};
+
+class FailableMemoryAllocator: public TestMemoryAllocator
+{
+public:
+    enum {MAX_NUMBER_OF_FAILED_ALLOCS = 10};
+    FailableMemoryAllocator(const char* name_str = "failable alloc", const char* alloc_name_str = "alloc", const char* free_name_str = "free");
+    virtual ~FailableMemoryAllocator() {}
+    virtual void failAllocNumber(int number);
+    virtual void failNthAllocAt(int allocationNumber, const char* file, int line);
```

Comment:
I like mine more :). failNthAllocAt makes it easier to remember that the allocation number is the first argument and location comes in the end. failAllocAt("file.cpp", 24, 2) makes it less explicit what 2 means. Always having to explicitely give the alloc number is not that bad.

But I guess it is because I like Zen of Python (explicit is better than implicit) and @basvodde likes Ruby :-).